keps/sig-storage/3314-csi-changed-block-tracking/README.md

```diff
@@ -0,0 +1,1656 @@
+- [Maturity levels (`alpha`, `beta`, `stable`)][maturity-levels]
+- [Feature gate][feature gate] lifecycle
+- [Deprecation policy][deprecation-policy]
+
+Clearly define what graduation means by either linking to the [API doc
+definition](https://kubernetes.io/docs/concepts/overview/kubernetes-api/#api-versioning)
+or by redefining what graduation means.
+
+In general we try to use the same stages (alpha, beta, GA), regardless of how the
+functionality is accessed.
+
+[feature gate]: https://git.k8s.io/community/contributors/devel/sig-architecture/feature-gates.md
+[maturity-levels]: https://git.k8s.io/community/contributors/devel/sig-architecture/api_changes.md#alpha-beta-and-stable-versions
+[deprecation-policy]: https://kubernetes.io/docs/reference/using-api/deprecation-policy/
+
+Below are some examples to consider, in addition to the aforementioned [maturity levels][maturity-levels].
+-->
+
+#### Alpha
+
+* Approval of the proposed CRDs and GRPC specification.
+* The `SnapshotMetadata` service can handle gRPC requests to get allocated and
+changed blocks, and stream responses back to client, without imposing load on
+the K8s API server.
+* Initial e2e tests completed and enabled.
+
+#### Beta
+
+* Involve 2 different storage providers and backup applications in the
+implementations of the `SnapshotMetadata` service to enable successful e2e
+backup workflow.
+* Increase e2e test coverage.
+* Gather feedback from CSI driver maintainers and backup users, especially
+on performance metrics.
+
+#### GA
+
+* CSI drivers include the `SnapshotMetadata` service as part of their distros.
+* Allowing time for user feedback.
+
+<!--
+**Note:** Generally we also wait at least two releases between beta and
+GA/stable, because there's no opportunity for user feedback, or even bug reports,
+in back-to-back releases.
+
+**For non-optional features moving to GA, the graduation criteria must include
+[conformance tests].**
+
+[conformance tests]: https://git.k8s.io/community/contributors/devel/sig-architecture/conformance-tests.md
+
+#### Deprecation
+
+- Announce deprecation and support policy of the existing flag
+- Two versions passed since introducing the functionality that deprecates the flag (to address version skew)
+- Address feedback on usage/changed behavior, provided on GitHub issues
+- Deprecate the flag
+-->
+
+### Upgrade / Downgrade Strategy
+
+<!--
+If applicable, how will the component be upgraded and downgraded? Make sure
+this is in the test plan.
+
+Consider the following in developing an upgrade/downgrade strategy for this
+enhancement:
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to maintain previous behavior?
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to make use of the enhancement?
+-->
+
+### Version Skew Strategy
```

Comment:
Have you given any thought to upgrade/downgrade/version skew? Presumably the new service / sidecar will be included with the CSI driver and versioned/upgraded along with that? Are there any special considerations around handling snapshots made with different versions? Presumably this is covered in the existing snapshot functionality? You may just want to describe that and provide pointers to existing processes and ways of doing upgrades.  And you may want to note any specific testing in this area you will add for this new service.